packages/grafana-data/src/transformations/transformers/calculateField.ts

```diff
@@ -68,56 +70,59 @@ export const calculateFieldTransformer: DataTransformerInfo<CalculateFieldTransf
       reducer: ReducerID.sum,
     },
   },
-  transformer: options => (data: DataFrame[]) => {
-    if (options && options.timeSeries !== false) {
-      data = ensureColumnsTransformer.transformer(null)(data);
-    }
-
-    const mode = options.mode ?? CalculateFieldMode.ReduceRow;
-    let creator: ValuesCreator | undefined = undefined;
-
-    if (mode === CalculateFieldMode.ReduceRow) {
-      creator = getReduceRowCreator(defaults(options.reduce, defaultReduceOptions), data);
-    } else if (mode === CalculateFieldMode.BinaryOperation) {
-      creator = getBinaryCreator(defaults(options.binary, defaultBinaryOptions), data);
-    }
-
-    // Nothing configured
-    if (!creator) {
-      return data;
-    }
-
-    return data.map(frame => {
-      // delegate field creation to the specific function
-      const values = creator!(frame);
-      if (!values) {
-        return frame;
-      }
+  transformer: (options, frames) => {
+    const source =
+      options && options.timeSeries !== false ? ensureColumnsTransformer.transformer(null, frames) : of(frames);
+
+    return source.pipe(
+      map(data => {
+        const mode = options.mode ?? CalculateFieldMode.ReduceRow;
```

Comment:
just a question/nit about types:
isn't `options.mode` already of type `CalculateFieldMode` (non-optional)?

if it is then maybe we can omit all checks on `undefined` for `creator` since `options.mode` con be either (and only) `ReduceRow` or `BinaryOperation`:


```ts
let creator: ValuesCreator;

if (options.mode === CalculateFieldMode.ReduceRow) {
  creator = getReduceRowCreator(defaults(options.reduce, defaultReduceOptions), data);
} else {
  creator = getBinaryCreator(defaults(options.reduce, defaultBinaryOptions), data);
}

return data.map(frame => {
  // delegate field creation to the specific function
  const values = creator(frame);
// [...]
```

Reply:
Thanks for feedback but I actually don't know because I didn't write this particular transformation but maybe @ryantxu can answer?